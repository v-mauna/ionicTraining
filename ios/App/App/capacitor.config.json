{
  "appId": "com.vanessamauna.weatherSharer",
  "appName": "Weather Sharer",
  "bundledWebRuntime": false,
  "npmClient": "npm",
  "webDir": "build",
  "plugins": {
    "SplashScreen": {
      "launchShowDuration": 0
    }
  },
  "cordova": {}
}
