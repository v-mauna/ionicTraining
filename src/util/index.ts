export * from './iconPaths';
export * from './weather';
export * from './settings';
