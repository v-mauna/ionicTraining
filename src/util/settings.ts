import { Plugins } from '@capacitor/core';

const { Storage } = Plugins;

class SettingsService {
  setScale(value: string): Promise<void> {}

  async getScale(): Promise<string> {}
}

export const settings = new SettingsService();
