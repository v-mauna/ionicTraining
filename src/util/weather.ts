import { Coordinate, Weather, UVIndex, WeeklyForecast} from '../models';

class WeatherService {
    private baseUrl = 'https://api.openweathermap.org/data/2.5';
    private appId = '0dd2e08fd19801e6177193ba30a52199';

    async current(): Promise<Weather> {
        const res = await this.fetchData('weather');
        const data = await res;
        return this.unpackWeather(data)
        };

    async forecast(): Promise<WeeklyForecast> {
        const res = await this.fetchData('forecast');
        const data = await res;
        return this.unpackForecast(data)
    };

    async uvIndex() : Promise<UVIndex> {
        const res = await this.fetchData('uvi');
        return this.unpackUVIndex(res)
    }

    private async fetchData(endpoint: string): Promise<any> {
        const loc = await this.currentLocation();
        const res = await fetch(`${this.baseUrl}/${endpoint}?lat=${loc.latitude}&lon=${loc.longitude}&appid=${this.appId}`);
        return res.json();
    }

    private async currentLocation(): Promise<Coordinate> {
        return Promise.resolve({
            latitude: 43.073051,
            longitude: -89.40123
        });
    }

    private unpackWeather (data: any): Weather {
        return {
            temperature: data.main.temp,
            condition: data.weather[0].id,
            date: data.dt && new Date(data.dt * 1000)
        };

    }  private unpackForecast(data: any): WeeklyForecast {
        let currentDay: Array<Weather>;
        let prevDate: number;
        const forecast: WeeklyForecast = [];

        data.list.forEach((item: any) => {
            const w = this.unpackWeather(item);
            if (w.date!.getDate() !== prevDate) {
                prevDate = w.date!.getDate();
                currentDay = [];
                forecast.push(currentDay);
            }
            currentDay.push(w);
        });

        return forecast;
    }

    private unpackUVIndex(data:any) : UVIndex {
        return {
            value: data.value,
            riskLevel:this.riskLevel(data.value)
        }
    }

    private riskLevel(value:number) : number {
       if(value < 3) return 0;
       if(value < 6)return 1;
       if(value < 8) return 2;
       if(value < 11) return 3;
       if(value >= 11) return 4;
       return 4

    }
}

export const weather = new WeatherService();