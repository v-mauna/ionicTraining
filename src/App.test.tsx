import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { weather } from './util';

it('renders without crashing', () => {
  const { baseElement } = render(<App />);
  expect(baseElement).toBeDefined();
});

it('contains three tabs', () => {
  const { container } = render(<App />);
  expect(container.querySelectorAll('ion-tab-button').length).toEqual(3);
});

it.each([
  [0, 'Current Weather'],
  [1, 'Forecast'],
  [2, 'UV Index']
])('contains the proper text for tab %i', (tab, text) => {
  const { container } = render(<App />);
  expect(container.querySelectorAll('ion-tab-button')[tab as number].textContent).toEqual(text);
});

it('renders consistently', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});


let spy: any;
beforeAll(() => {
  spy = jest.spyOn(weather, 'current').mockImplementation(() =>
      Promise.resolve({
        temperature: 280.32,
        condition: 300,
        date: new Date(1485789600 * 1000),
      } as any),
  );
});

afterAll(() => spy.mockReset());