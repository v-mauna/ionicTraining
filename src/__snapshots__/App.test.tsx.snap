// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders consistently 1`] = `
<DocumentFragment>
  <ion-app>
    <div
      class="ion-page "
    >
      <ion-tabs
        classname="ion-tabs"
        style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
      >
        <div
          class="tabs-inner"
          style="position: relative; flex: 1; contain: layout size style;"
        >
          <ion-router-outlet />
        </div>
        <ion-tab-bar
          selected-tab="current-weather"
          slot="bottom"
        >
          <ion-tab-button
            href="/"
            tab="current-weather"
          >
            <ion-icon
              icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M464 464H48a16 16 0 01-14.07-23.62l208-384a16 16 0 0128.14 0l208 384A16 16 0 01464 464z'/></svg>"
            />
            <ion-label>
              Current Weather
            </ion-label>
          </ion-tab-button>
          <ion-tab-button
            href="/forecast"
            tab="forecast"
          >
            <ion-icon
              icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M256 464c-114.69 0-208-93.31-208-208S141.31 48 256 48s208 93.31 208 208-93.31 208-208 208z'/></svg>"
            />
            <ion-label>
              Forecast
            </ion-label>
          </ion-tab-button>
          <ion-tab-button
            href="/uv-index"
            tab="uv-index"
          >
            <ion-icon
              icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M416 464H96a48.05 48.05 0 01-48-48V96a48.05 48.05 0 0148-48h320a48.05 48.05 0 0148 48v320a48.05 48.05 0 01-48 48z'/></svg>"
            />
            <ion-label>
              UV Index
            </ion-label>
          </ion-tab-button>
        </ion-tab-bar>
      </ion-tabs>
    </div>
  </ion-app>
</DocumentFragment>
`;
